streams/lms_metadata.py

```diff
@@ -0,0 +1,218 @@
+        return True #If the hostname is already an IP, return True
+      else:
+        host = socket.gethostbyname(self.server)
+        if re.match(ip_regex, host):
+          logging.debug(f"hostname: {self.server}")
+          logging.debug(f"ip: {host}")
+          self.server = host
+          return True #If the resolved hostname is an IP, return True
+        else:
+          return False #If the hostname is resolvable but somehow still not an IP, return False
+
+    except Exception:
+      logging.warning("Could not resolve hostname, trying to find LMS server manually")
+      logging.warning("Set local network DNS settings or set 'server' in config to be an IP to avoid this issue in the future")
+      return False #If there is an error, or the hostname is unresolvable, return false
+
+  def connect(self):
+    """Discovers LMS Player and then requests metadata repetitively"""
+    connected = False
+    self.meta.save_file(self.folder)
+
+    # When not connected, search for player to connect to by the proper name
+    abort = False # Used to stop looping without initiating the secondary loop
+    while not connected and not abort:
+      try:
+        if not self.resolve_host(): # Attempt to resolve hostname, if unresolvable then find host with find_lms_server script
+          machine = platform.machine()
+          logging.debug(f"platform.machine() output: {machine}")
+
+          find_lms_server = None
+          if machine == "x86_64":
+            find_lms_server = "bin/x64/find_lms_server"
+          elif machine == "armv7l":
+            find_lms_server = "bin/arm/find_lms_server"
+          else:
+            self.meta.artist = "Unsupported CPU architecture for LMS MetaData"
+            self.meta.album = "Please set 'server' in LMS config to an IP"
+            self.meta.track = "or contact AmpliPi Support:support@micro-nova.com"
+            logging.error("LMS metadata reader has detected an unsupported chipset")
+            logging.warning("Aborting LMS metadata search")
+            abort = True
+
+          # Much faster method of connecting to the metadata server using code from: https://github.com/ralph-irving/squeezelite/blob/master/tools/find_server.c
+          # faster relative to the original method, using NMAP to go door to door (ip to ip) and ask if self.name is home
+          ip_find = subprocess.run([find_lms_server], check=True, capture_output=True, text=True)
+          # Uses regex because ip_find spits out as '{Hostname}:{port} ({ip})'
+          self.server = re.search(r'\((.*?)\)', ip_find.stdout).group(1)
+
+        if self.debug:
+          logging.debug(f"Found LMS Server: {self.server}:{self.port}")
+
+        player_json = {"id": 1,	"method": "slim.request",	"params": [self.player_name, ["players", "-", 100, "playerid"]]}
+        player_info = requests.get(f'http://{self.server}:{self.port}/jsonrpc.js', json=player_json, timeout=5)
```

Comment:
![image](https://github.com/micro-nova/AmpliPi/assets/114415937/875bbecd-1a3a-416a-b30a-a44287bd9bac)

For some reason if you don't populate self.server during __init__, this has a chance of never getting anything for that value and so polls for `http://:900/jsonrpc.js` which is a fail state

self.server should always be populated by this point though, if it wasn't solved by `resolve_host` then it should be solved by `find_lms_server`; if somehow neither of those happen it should error out before ever reaching this point in the code